File: .agents/skills/bio-data-visualization-circos-plots/examples/circos_basic.R
#!/usr/bin/env Rscript
# Basic circos plot with circlize

library(circlize)

# Initialize with human genome
circos.initializeWithIdeogram(species = 'hg38', plotType = c('axis', 'labels'))

# Create example data
set.seed(42)
bed <- data.frame(
    chr = paste0('chr', sample(1:22, 200, replace = TRUE)),
    start = sample(1:1e8, 200),
    value = runif(200)
)
bed$end <- bed$start + 1e6

# Add scatter track
circos.genomicTrack(bed, panel.fun = function(region, value, ...) {
    circos.genomicPoints(region, value, pch = 16, cex = 0.5, col = 'steelblue')
}, track.height = 0.1)

# Add histogram track
circos.genomicTrack(bed, panel.fun = function(region, value, ...) {
    circos.genomicRect(region, value, ytop.column = 1, ybottom = 0,
                       col = 'coral', border = NA)
}, track.height = 0.1)

# Add links between regions
link_data <- data.frame(
    chr1 = c('chr1', 'chr3', 'chr7'),
    start1 = c(1e7, 5e7, 8e7),
    end1 = c(2e7, 6e7, 9e7),
    chr2 = c('chr5', 'chr10', 'chr15'),
    start2 = c(3e7, 4e7, 2e7),
    end2 = c(4e7, 5e7, 3e7)
)

for (i in seq_len(nrow(link_data))) {
    circos.link(link_data$chr1[i], c(link_data$start1[i], link_data$end1[i]),
                link_data$chr2[i], c(link_data$start2[i], link_data$end2[i]),
                col = adjustcolor('purple', alpha.f = 0.5))
}

circos.clear()

# Save to PDF
pdf('circos_output.pdf', width = 10, height = 10)
circos.initializeWithIdeogram(species = 'hg38', plotType = c('axis', 'labels'))
circos.genomicTrack(bed, panel.fun = function(region, value, ...) {
    circos.genomicPoints(region, value, pch = 16, cex = 0.5, col = 'steelblue')
}, track.height = 0.1)
circos.clear()
dev.off()
cat('Saved circos_output.pdf\n')

File: .agents/skills/bio-data-visualization-circos-plots/examples/circos_basic.py
# COPYRIGHT NOTICE
# This file is part of the "Universal Biomedical Skills" project.
# Copyright (c) 2026 MD BABU MIA, PhD <md.babu.mia@mssm.edu>
# All Rights Reserved.
#
# This code is proprietary and confidential.
# Unauthorized copying of this file, via any medium is strictly prohibited.
#
# Provenance: Authenticated by MD BABU MIA

#!/usr/bin/env python3
'''Basic circos plot with pyCircos'''

from pycircos import Gcircle, Garc
import numpy as np

chromosomes = [
    ('chr1', 248956422), ('chr2', 242193529), ('chr3', 198295559),
    ('chr4', 190214555), ('chr5', 181538259), ('chr6', 170805979),
    ('chr7', 159345973), ('chr8', 145138636), ('chr9', 138394717),
    ('chr10', 133797422), ('chr11', 135086622), ('chr12', 133275309)
]

circle = Gcircle()

for name, length in chromosomes:
    arc = Garc(arc_id=name, size=length, interspace=3, raxis_range=(850, 900),
               labelposition=60, label_visible=True)
    circle.add_garc(arc)

circle.set_garcs()

# Add scatter track
for name, length in chromosomes:
    positions = np.random.randint(0, length, 30)
    values = np.random.random(30)
    circle.scatterplot(name, data=values, positions=positions,
                       raxis_range=(700, 800), facecolor='steelblue', markersize=4)

# Add bar track
for name, length in chromosomes:
    positions = np.linspace(0, length, 50, dtype=int)
    values = np.random.random(50) * 100
    circle.barplot(name, data=values, positions=positions,
                   raxis_range=(550, 680), facecolor='coral')

# Add a link between regions
circle.chord_plot(('chr1', 50000000, 60000000), ('chr5', 100000000, 110000000),
                  raxis_range=(0, 500), facecolor='purple', alpha=0.5)

fig = circle.figure
fig.savefig('circos_output.png', dpi=300, bbox_inches='tight')
print('Saved circos_output.png')

__AUTHOR_SIGNATURE__ = "9a7f3c2e-MD-BABU-MIA-2026-MSSM-SECURE"
